compiler/modules/bir/bir.bal

```diff
@@ -592,22 +592,17 @@ public type EqualityInsn readonly & record {|
 # any function call could result in a stack overflow panic.
 # XXX This does not handle functions that don't return
 # (i.e. with return type of never)
+# XXX: This can also panic due to memory allocation for uniform function call
+# which is not handled gracefully
 public type CallInsn readonly & record {|
     *ResultInsnBase;
     # Position in the source that resulted in the instruction
     INSN_CALL name = INSN_CALL;
-    FunctionRef func;
-    Operand[] args;
-|};
-
-# Call a function using a function value.
-# This behaves similar to CallInsn.
-# XXX: In addition this can also panic due to memory allocation for uniform function call
-# which is not handled gracefully
-public type CallIndirectInsn readonly & record {|
-    *ResultInsnBase;
-    INSN_CALL_INDIRECT name = INSN_CALL_INDIRECT;
-    [Register, Operand...] operands;
+    [FunctionOperand, Operand...] operands;
+    # This indicates whether var args have been accumulated into a single list.
+    # If so the last operand is expected to be that list. This is possible only
```

Comment:
Need to fix this comment to use terminology of the spec.

Reply:
Fixed